hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdiff/RocksDBCheckpointDiffer.java

```diff
@@ -471,13 +529,151 @@ public HashSet<String> readRocksDBLiveFiles(String dbPathArg) {
+  /**
+   * Read compaction log until all dest (and src) db checkpoint SST
+   * nodes show up in the graph, or when it reaches the end of the log.
+   */
+  private boolean loadCompactionDAGBySSTSet(HashSet<String> sstSet) {
+
+    // Make a copy of the sstSet
+    HashSet<String> loadSet = new HashSet<>(sstSet);
+
+    // Check if all the nodes in the provided SST set is already loaded in DAG
+    for (String sstFile : sstSet) {
+      if (compactionNodeTable.containsKey(sstFile)) {
+        loadSet.remove(sstFile);
+      }
+    }
+
+    if (loadSet.size() == 0) {
+      // All expected nodes in the sstSet are already there,
+      //  no need to read any compaction log
+      return true;
+    }
+
+    // Otherwise, load compaction logs in order until all nodes are present in
+    //  the DAG.
+    try {
+      try (Stream<Path> pathStream = Files.list(Paths.get(compactionLogDir))
+          .filter(e -> e.toString().toLowerCase().endsWith(".log"))
+          .sorted()) {
+        for (Path logPath : pathStream.collect(Collectors.toList())) {
+
+          // TODO: Potential optimization: stop reading as soon as all nodes are
+          //  there. Currently it loads an entire file at a time.
+          readCompactionLogToDAG(logPath.toString());
+
+          for (Iterator<String> it = loadSet.iterator(); it.hasNext();) {
+            String sstFile = it.next();
+            if (compactionNodeTable.containsKey(sstFile)) {
+              LOG.debug("Found SST node: {}", sstFile);
+              it.remove();
+            }
+          }
+
+          if (loadSet.size() == 0) {
+            break;
+          }
+        }
+      }
+    } catch (IOException e) {
+      throw new RuntimeException("Error listing compaction log dir " +
+          compactionLogDir, e);
+    }
+
+    // Just in case there are still nodes to be expected not loaded.
+    if (loadSet.size() > 0) {
+      LOG.warn("The following nodes are missing from the compaction log: {}. "
+          + "Possibly because those a newly flushed SSTs that haven't gone "
+          + "though any compaction yet", loadSet);
+      return false;
+    }
+
+    return true;
+  }
+
+  /**
+   * Given the src and destination Snapshots, it prints a Diff list.
+   *
+   * Expected input: src is a checkpoint taken AFTER dest checkpoint.
+   *
+   * @param src
+   * @param dest
+   * @throws RocksDBException
+   */
+  private synchronized List<String> printSnapdiffSSTFiles(
+      Snapshot src, Snapshot dest) {
+
+    LOG.warn("src db checkpoint: {}", src.dbPath);  // from
     HashSet<String> srcSnapFiles = readRocksDBLiveFiles(src.dbPath);
-    LOG.warn("dest Snapshot files :" + dest.dbPath);
+    LOG.warn("dest db checkpoint: {}", dest.dbPath);  //to
     HashSet<String> destSnapFiles = readRocksDBLiveFiles(dest.dbPath);
+    System.out.println();
+
+    // Read compaction log until all dest (and src) db checkpoint SST
+    // nodes show up in the graph
+//    loadCompactionDAGBySSTSet(destSnapFiles);
```

Comment:
Is this line commented out intentionally? we should load destSnapFiles also right?

Reply:
Yes this is intentional because in reality it is redundant. When loading `srcSnapFiles`, `destSnapFiles` should have been traversed already under the current assumption that:

> Expected input: src is a checkpoint taken AFTER dest checkpoint

Reply:
Got it, we can maybe have a preconditions check to validate the generation for src and dest.

Reply:
Done